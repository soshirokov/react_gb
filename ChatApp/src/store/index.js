import {createStore, combineReducers} from 'redux';
import { chatReducer } from './chats/reducer';
import {profileReducer} from './profile/reducer';

const reducers = combineReducers({
    profile: profileReducer,
    chats: chatReducer
});

export const store = createStore(
    reducers, 
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
    );   