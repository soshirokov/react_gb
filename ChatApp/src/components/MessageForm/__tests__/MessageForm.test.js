import { MessageForm } from '..';
import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import { Provider } from 'react-redux';
import { store } from '../../../store';

describe('test MessageForm presentation', () => {
    it("MessageForm matches snaphot", () => {
        const result = render(<Provider store={store}><MessageForm addMessage={()=>{}}/></Provider>);

        expect(result).toMatchSnapshot();
    });

    it('calls addMessage when form submit', () => {
        const myHandler = jest.fn();

        render(<Provider store={store}><MessageForm addMessage={myHandler}/></Provider>);

        const submitBtn = screen.getByText('Send');

        fireEvent(
            submitBtn,
            new MouseEvent("click", {
              bubbles: true,
              cancelable: true,
            })
        );

        expect(myHandler).toHaveBeenCalledTimes(1);
    });
});