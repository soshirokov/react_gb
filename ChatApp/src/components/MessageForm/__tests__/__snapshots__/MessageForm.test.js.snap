// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test MessageForm presentation MessageForm matches snaphot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="form"
      >
        <form>
          <div
            class="MuiFormControl-root MuiTextField-root form__input css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for="standard-basic"
              id="standard-basic-label"
            >
              Type something
            </label>
            <div
              class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
                id="standard-basic"
                type="text"
                value=""
              />
            </div>
          </div>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root form__submit css-sghohy-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="submit"
          >
            Send
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="form"
    >
      <form>
        <div
          class="MuiFormControl-root MuiTextField-root form__input css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="standard-basic"
            id="standard-basic-label"
          >
            Type something
          </label>
          <div
            class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl css-1480iag-MuiInputBase-root-MuiInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInput-input MuiInputBase-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
              id="standard-basic"
              type="text"
              value=""
            />
          </div>
        </div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root form__submit css-sghohy-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Send
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
